compiler/native/expand.go

```diff
@@ -48,6 +49,12 @@ func (c *client) ExpandSteps(s yaml.StepSlice, tmpls map[string]*yaml.Template)
 			continue
 		}
 
+		// inject environment information for template
+		step, err := c.EnvironmentStep(step)
```

Comment:
Curious around the purpose of this 👍 

It appears like we're injecting the env vars for a templated step before we render the template later on?

i.e. something like this https://github.com/go-vela/community/issues/72

Am I reading this correctly? Or am I misunderstanding what we're doing this for?

Reply:
Injection of environment variables is needed to automatically inject system/platform variables into the template as variables underneath ctx["vela"], the actual code implementation for that can be found [here](https://github.com/go-vela/compiler/pull/90/files#diff-b25ebe915de5f17f1addab7c5a2be4ffc10c1cddd34d6f0c179ae899453bdbe7R65). You can find one sample of utilizing that syntax [here](https://github.com/go-vela/compiler/pull/90/files#diff-37b270bc3de86b6a6bb51d3608cbae9a3b113400fc00d0b0ea0189c6b854a0ecR5).

Reply:
Should be fairly straightforward to solve https://github.com/go-vela/community/issues/72 for Go templates in another PR.

Reply:
You are correct, that's exactly what it's doing. It's injecting those build-time variables into the step so they can be used within the template if a user wants to reference them. We didn't add them into the Go template to be accessible yet but it's one step closer to allowing that behavior. The Go template package would just need to be updated to use those variables it now has access to.